SingularityService/src/main/java/com/hubspot/singularity/data/history/DeployTaskHistoryHelper.java

```diff
@@ -0,0 +1,112 @@
+  }
+
+  public List<SingularityTaskIdHistory> getHistoriesFor(Collection<SingularityTaskId> taskIds) {
+    Map<SingularityTaskId, SingularityTask> tasks = taskManager.getTasks(taskIds);
+    Map<SingularityTaskId, List<SingularityTaskHistoryUpdate>> map = taskManager.getTaskHistoryUpdates(taskIds);
+
+    List<SingularityTaskIdHistory> histories = Lists.newArrayListWithCapacity(taskIds.size());
+
+    for (SingularityTaskId taskId : taskIds) {
+      List<SingularityTaskHistoryUpdate> historyUpdates = map.get(taskId);
+      SingularityTask task = tasks.get(taskId);
+
+      histories.add(SingularityTaskIdHistory.fromTaskIdAndTaskAndUpdates(taskId, task, historyUpdates));
+    }
+
+    Collections.sort(histories);
+
+    return histories;
+  }
+
+  @Override
+  protected List<SingularityTaskIdHistory> getFromZk(final SingularityDeployKey deployKey) {
+    List<SingularityTaskId> requestTaskIds = taskManager.getTaskIdsForRequest(deployKey.getRequestId());
+
+    final Iterable<SingularityTaskId> deployTaskIds = Iterables.filter(requestTaskIds, new Predicate<SingularityTaskId>() {
+      @Override
+      public boolean apply(SingularityTaskId input) {
+        return input.getDeployId().equals(deployKey.getDeployId());
+      }
+    });
+
+    return getHistoriesFor(ImmutableList.copyOf(deployTaskIds));
+  }
+
+  @Override
+  protected List<SingularityTaskIdHistory> getFromHistory(final SingularityDeployKey deployKey, int historyStart, int numFromHistory) {
+    List<SingularityTaskIdHistory> requestHistories = historyManager.getTaskHistoryForRequest(deployKey.getRequestId(), historyStart, numFromHistory);
+
+    final Iterable<SingularityTaskIdHistory> deployHistories = Iterables.filter(requestHistories, new Predicate<SingularityTaskIdHistory>() {
+      @Override
+      public boolean apply(SingularityTaskIdHistory input) {
+        return input.getTaskId().getDeployId().equals(deployKey.getDeployId());
+      }
+    });
+
+    return ImmutableList.copyOf(deployHistories);
+  }
+
+  public List<SingularityTaskIdHistory> getActiveDeployTasks(SingularityDeployKey key) {
+    List<SingularityTaskIdHistory> histories = this.getFromZk(key);
+
+    final Iterable<SingularityTaskIdHistory> activeHistories = Iterables.filter(histories, new Predicate<SingularityTaskIdHistory>() {
+      @Override
+      public boolean apply(SingularityTaskIdHistory input) {
+        return !input.getLastTaskState().get().isDone();
+      }
+    });
+
+    return ImmutableList.copyOf(activeHistories);
+  }
+
+  public List<SingularityTaskIdHistory> getInactiveDeployTasks(SingularityDeployKey key, Integer limitCount, Integer limitStart) {
+    // We don't know our limits yet before filtering task state
+    Integer limit = (int) (singularityConfiguration.getHistoryPurgingConfiguration().getDeleteTaskHistoryAfterTasksPerRequest().or(10000) * 1.2);
+    List<SingularityTaskIdHistory> histories = this.getBlendedHistory(key, 0, limit);
```

Comment:
going to add a migration (in new pr) that will give us a deployId column, that will make things much cleaner since right now we can't find tasks for a deploy without looking at a much larger number of items then filtering